tests/ValueParsers/TimeParserTest.php

```diff
@@ -461,6 +493,13 @@ public function invalidInputProvider() {
 			array(),
 			'foooooooooo',
 			'1 June 2014',
+			'59-01-01',
+			'+59-01-01',
```

Comment:
As the comment says: Either the first number is a year for sure (which is only the case if it's bigger than 59) or the "T" part is present. Incomplete ISO strings like these can be a time or in an order different from YMD. We have an other parser for such cases. A more obvious example may be "12-11-10". I'm using 59 as a limit instead of 31 just to be on the safe side.